src/internal/list.js

```diff
@@ -0,0 +1,18 @@
+function List(head, tail){
+  this.head = head;
+  this.tail = tail;
+  this.size = tail.size + 1;
+}
+
+List.prototype.isEmpty = false;
+
+List.prototype.cons = function List$cons(x){
+  return new List(x, this);
+};
+
+const empty = Object.create(List.prototype);
+
+empty.isEmpty = true;
+empty.size = 0;
+
+export default empty;
```

Comment:
@davidchambers I'm still quite fond of exporting `empty` from `list`. Essentially we're exporting `[]` and `:` (`cons`), and those are all the necessary building blocks for creating lists. Not only do I feel like this is elegant in a way, it's also quite convenient because I don't have to do anything before using it the way I want to in `core.js`.